pallets/staking/src/benchmarking.rs

```diff
@@ -63,36 +71,42 @@ pub fn create_validator_with_nominators<T: Trait>(
 pub fn create_validator_with_nominators_with_balance<T: Trait>(
     n: u32,
     upper_bound: u32,
-    balance: BalanceOf<T>,
+    balance: u32,
     dead: bool,
-) -> Result<T::AccountId, &'static str> {
+) -> Result<T::AccountId, DispatchError> {
     let mut points_total = 0;
     let mut points_individual = Vec::new();
 
     let (v_stash, v_controller) = create_stash_controller_with_balance::<T>(0, balance)?;
-    let v_controller_origin = RawOrigin::Signed(v_controller.clone());
+    let v_controller_origin = v_controller.origin();
 
     let validator_prefs = ValidatorPrefs {
-        commission: Perbill::from_percent(10),
+        commission: Perbill::from_percent(50),
     };
-    Staking::<T>::validate(v_controller_origin.into(), validator_prefs)?;
-    let stash_lookup: <T::Lookup as StaticLookup>::Source = T::Lookup::unlookup(v_stash.clone());
+    emulate_validator_setup::<T>(1, 10, Perbill::from_percent(60));
+    Staking::<T>::add_permissioned_validator(RawOrigin::Root.into(), v_stash.did(), Some(2))
+        .expect("Failed to add permissioned validator");
+    Staking::<T>::validate(v_controller_origin.into(), validator_prefs.clone())
+        .expect("Failed to validate");
+    assert_eq!(
+        Staking::<T>::validators(v_stash.account()),
+        validator_prefs,
+        "Failed to set the validator"
+    );
+    let stash_lookup = v_stash.lookup();
 
     points_total += 10;
-    points_individual.push((v_stash.clone(), 10));
+    points_individual.push((v_stash.account(), 10));
 
     // Give the validator n nominators, but keep total users in the system the same.
     for i in 0..upper_bound {
         let (_n_stash, n_controller) = if !dead {
-            create_stash_controller::<T>(u32::max_value() - i, 100)?
+            create_stash_controller_with_balance::<T>(u32::max_value() - i, 100)?
         } else {
-            create_stash_and_dead_controller::<T>(u32::max_value() - i, 100)?
+            create_stash_controller::<T>(u32::max_value() - i, 100)?
```

Comment:
This creates a controller with balance of 100. We want dead/no balance controller.

Reply:
This isn't fixed either. Albeit the Controller has 0 balance, the rewards destination is set as staked instead of controller. That defeats the purpose of having zero balance controller. Please change the rewards destination in `create_stash_controller` fn to be the controller.

Reply:
@satyamakgec This is still not fixed. Even with 0 balance, you are calling user builder which calls `make_free_balance_be` and that creates the storage element even though with 0 value. By dead controller, we want to test the scenario where the balance does not exist.

Reply:
@maxsam4 please verify now the behavior.